src/Components/Web/src/StreamRenderingAttribute.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
```

Comment:
Open question for API review: should this actually go in the `Microsoft.AspNetCore.Components.Web` namespace?

I've put it in this assembly because it needs to be usable in an RCL that might have no other direct dependency on SSR concepts (so it definitely can't go in the `.Endpoints` assembly). However I didn't put it in `M.A.Components` because I'm not aware of any case where someone would be writing a web-targeted RCL without referencing the `M.A.C.Web` assembly. If in the future we did want to move this to `M.A.Components` we could do so by type-forwarding, but at that point would still be stuck with whatever namespace is selected here (hence perhaps it's better to be in the `M.A.Components` namespace now, not `.Web`).

Reply:
> I don't think we've materialized the benefit of having a split between Components and Web

Yeah I agree that's largely true, however if this framework is going to stick around for many years to come, it's completely possible that at some point there will be a goal to have a non-web use case (like Mobile Blazor Bindings) in which case the split would become super valuable. Even if not I've found it helpful to clarify the architecture and pay attention to what's the essence of the component model and what's its application to the web.